test/mpi/test_allgatherv.cc

```diff
@@ -64,7 +61,11 @@ TestAllgatherv::TestAllgatherv(size_t _msgsize, ucc_test_mpi_inplace_t _inplace,
     args.dst.info_v.displacements = (ucc_aint_t*)displacements;
     args.dst.info_v.datatype      = TEST_DT;
     args.dst.info_v.mem_type      = _mt;
-    UCC_CHECK(ucc_collective_init(&args, &req, team.team));
+
+    UCC_CHECK_SKIP(ucc_collective_init(&args, &req, team.team), test_skip);
```

Comment:
Looks like now the test will init the collective if max mssg size is exceeded - so it won't be skipped, because you do collective_init before the skip_reduce. If we exceeding the msg size we don't want to spend time allocating/initializing/freeing buffers.

Reply:
Fixed, but it adds extra mpi_allreduce for the skip_cause

Reply:
@vspetrov removed mpi_allreduce for after `ucc_collective_init`